Add missing color interpolation methods added to CSS color since initial implementation

weinig opened the pull request:
#### 8a3d3672b571656a923a0ed60872d771ac33b133
<pre>
Add missing color interpolation methods added to CSS color since initial implementation
<a href="https://bugs.webkit.org/show_bug.cgi?id=272646">https://bugs.webkit.org/show_bug.cgi?id=272646</a>

Reviewed by Tim Nguyen.

Adds support for the four additional color interpolation methods,
`display-p3`, `a98-rgb`, `prophoto-rgb` and `rec2020`, that have
been added as valid interpolation method color spaces since the
initial implementation.

Tests updated:
    imported/w3c/web-platform-tests/css/css-color/parsing/color-computed-color-mix-function.html
    imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-color-mix-function.html

* Source/WebCore/css/CSSGradientValue.cpp:
(WebCore::appendColorInterpolationMethod):
(WebCore::appendHueInterpolationMethod): Deleted.
Refactor to re-use more of the existing serialization in ColorInterpolationMethod.h
and reduce duplication by using a generic overload.

* Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp:
(WebCore::CSSPropertyParserHelpers::consumeColorInterpolationMethod):
Adds support for parsing the new color spaces. Also fixes the comment
with the spec text to remove &quot;specified&quot; as a hue-interpolation-method
as that is no longer supported (both in the spec and WebCore).

* Source/WebCore/platform/graphics/ColorInterpolationMethod.cpp:
(WebCore::serializationForCSS):
(WebCore::operator&lt;&lt;):
* Source/WebCore/platform/graphics/ColorInterpolationMethod.h:
(WebCore::serializationForCSS):
Add new interpolation method color spaces and their serialization.
Expose a few serialization helpers so that the gradient serialization
can use them as well.

* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
Update to match new definitions.

Canonical link: <a href="https://commits.webkit.org/277481@main">https://commits.webkit.org/277481@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/cfd010f14bce22cd2c9b3ba7ad0e3826021ac357

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/47744 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/26933 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/50509 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/50427 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/43799 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/50051 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/32789 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/24413 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/38858 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/48326 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/24586 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/41223 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/20155 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/22067 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/42414 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/5793 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/44090 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/42837 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/52321 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/22780 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/19109 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/46163 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/24052 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/45198 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10532 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/24840 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/23772 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m reviewed (approved):
Would be good to land with WPT rebaselines once the import lands

Commit:
Add missing color interpolation methods added to CSS color since initial implementation
https://bugs.webkit.org/show_bug.cgi?id=272646

Reviewed by Tim Nguyen.

Adds support for the four additional color interpolation methods,
`display-p3`, `a98-rgb`, `prophoto-rgb` and `rec2020`, that have
been added as valid interpolation method color spaces since the
initial implementation.

Tests updated:
    imported/w3c/web-platform-tests/css/css-color/parsing/color-computed-color-mix-function.html
    imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-color-mix-function.html

* Source/WebCore/css/CSSGradientValue.cpp:
(WebCore::appendColorInterpolationMethod):
(WebCore::appendHueInterpolationMethod): Deleted.
Refactor to re-use more of the existing serialization in ColorInterpolationMethod.h
and reduce duplication by using a generic overload.

* Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp:
(WebCore::CSSPropertyParserHelpers::consumeColorInterpolationMethod):
Adds support for parsing the new color spaces. Also fixes the comment
with the spec text to remove "specified" as a hue-interpolation-method
as that is no longer supported (both in the spec and WebCore).

* Source/WebCore/platform/graphics/ColorInterpolationMethod.cpp:
(WebCore::serializationForCSS):
(WebCore::operator<<):
* Source/WebCore/platform/graphics/ColorInterpolationMethod.h:
(WebCore::serializationForCSS):
Add new interpolation method color spaces and their serialization.
Expose a few serialization helpers so that the gradient serialization
can use them as well.

* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
Update to match new definitions.

Canonical link: https://commits.webkit.org/277481@main
